vader compile issue

As reported by @siegmargross in https://www.mail-archive.com/users@lists.open-mpi.org/msg32704.html, he's having a compile issue with the vader BTL.

@hjelmn can you have a look?

```
oki openmpi-master-201809290304-73075b8-Linux.x86_64.64_cc 151 head -7 config.log | tail -1
 $ ../openmpi-master-201809290304-73075b8/configure --prefix=/usr/local/openmpi-master_64_cc --libdir=/usr/local/openmpi-master_64_cc/lib64 --with-jdk-bindir=/usr/local/jdk-10.0.1/bin --with-jdk-headers=/usr/local/jdk-10.0.1/include JAVA_HOME=/usr/local/jdk-10.0.1 LDFLAGS=-m64 -mt -Wl,-z -Wl,noexecstack -L/usr/local/lib64 CC=cc CXX=CC FC=f95 CFLAGS=-m64 -mt CXXFLAGS=-m64 FCFLAGS=-m64 CPP=cpp CXXCPP=cpp --disable-mpi-fortran --enable-mpi-cxx --enable-cxx-exceptions --enable-mpi-java --with-valgrind=/usr/local/valgrind --with-hwloc=internal --without-verbs --with-wrapper-cflags=-std=c11 -m64 -mt --with-wrapper-cxxflags=-m64 --with-wrapper-fcflags=-m64 --with-wrapper-ldflags=-mt --enable-debug


loki openmpi-master-201809290304-73075b8-Linux.x86_64.64_cc 151 tail -33 log.make.Linux.x86_64.64_cc
 CC       btl_vader_sendi.lo
 CC       btl_vader_get.lo
"../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_fifo.h", line 157: operand cannot have void type: op "="
"../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_fifo.h", line 157: assignment type mismatch:
       long "=" void
cc: acomp failed for ../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_sendi.c
Makefile:1929: recipe for target 'btl_vader_sendi.lo' failed
make[2]: *** [btl_vader_sendi.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
"../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_fifo.h", line 157: operand cannot have void type: op "="
"../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_fifo.h", line 157: assignment type mismatch:
       long "=" void
cc: acomp failed for ../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_send.c
"../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_fifo.h", line 157: operand cannot have void type: op "="
"../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_fifo.h", line 157: assignment type mismatch:
       long "=" void
Makefile:1929: recipe for target 'btl_vader_send.lo' failed
make[2]: *** [btl_vader_send.lo] Error 1
cc: acomp failed for ../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_module.c
Makefile:1929: recipe for target 'btl_vader_module.lo' failed
make[2]: *** [btl_vader_module.lo] Error 1
"../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_fifo.h", line 157: operand cannot have void type: op "="
"../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_fifo.h", line 157: assignment type mismatch:
       long "=" void
cc: acomp failed for ../../../../../openmpi-master-201809290304-73075b8/opal/mca/btl/vader/btl_vader_component.c
Makefile:1929: recipe for target 'btl_vader_component.lo' failed
make[2]: *** [btl_vader_component.lo] Error 1
make[2]: Leaving directory '/export2/src/openmpi-master/openmpi-master-201809290304-73075b8-Linux.x86_64.64_cc/opal/mca/btl/vader'
Makefile:2392: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/export2/src/openmpi-master/openmpi-master-201809290304-73075b8-Linux.x86_64.64_cc/opal'
Makefile:1910: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
loki openmpi-master-201809290304-73075b8-Linux.x86_64.64_cc 152
```

Siegmar specifically stated that he tested master, but I'm *guessing* that this affects all of 2.1.x, 3.0.x, 3.1.x, and 4.0.x.

--- a/opal/mca/btl/vader/btl_vader_fifo.h
+++ b/opal/mca/btl/vader/btl_vader_fifo.h
@@ -12,7 +12,7 @@
  *                         All rights reserved.
  * Copyright (c) 2006-2007 Voltaire. All rights reserved.
  * Copyright (c) 2009-2010 Cisco Systems, Inc.  All rights reserved.
- * Copyright (c) 2010-2017 Los Alamos National Security, LLC.
+ * Copyright (c) 2010-2018 Los Alamos National Security, LLC.
  *                         All rights reserved.
  * $COPYRIGHT$
  *
@@ -154,7 +154,11 @@ static inline mca_btl_vader_hdr_t *vader_fifo_read (vader_fifo_t *fifo, struct m
 
 static inline void vader_fifo_init (vader_fifo_t *fifo)
 {
-    fifo->fifo_head = fifo->fifo_tail = VADER_FIFO_FREE;
+    /* due to a compiler bug in Oracle C 5.15 the following line was broken into two. Not
+     * ideal but oh well. See #5814 */
+    /* fifo->fifo_head = fifo->fifo_tail = VADER_FIFO_FREE; */
+    fifo->fifo_head = VADER_FIFO_FREE;
+    fifo->fifo_tail = VADER_FIFO_FREE;
     fifo->fbox_available = mca_btl_vader_component.fbox_max;
     mca_btl_vader_component.my_fifo = fifo;
 }

